webapp/src/components/Analysis/UtterancesTable.tsx

```diff
@@ -390,6 +399,37 @@ const UtterancesTable: React.FC<Props> = ({
   ];
 
   const searchString = constructSearchString(pipeline);
+
+  const importProposedActions = (file: File) => {
+    const fileReader = new FileReader();
+    fileReader.onload = ({ target }) => {
+      if (target) {
+        const result = target.result as string;
+        const [header, ...rows] = result.split(/\r?\n/).slice(0, -1);
```

Comment:
I'm sorry for changing my mind again, but the more I think about it, the more I like my original suggestion of trimming the whole `result` before splitting. The current approach of discarding the last row with `.slice(0, -1)` would silently lose a row if there were no newline at the end of the file, which might happen if the user manually edits the file, and would result in losing data silently. I'd use `.trimEnd()` as I don't expect extra newlines at the beginning of the file, only at the end. Does that make sense?
```suggestion
        const [header, ...rows] = result.trimEnd().split(/\r?\n/);
```

Reply:
Otherwise, we should check if there is any new line at the end to do a slice(0,-1).